examples/isolated_declarations/backend/BUILD.bazel

```diff
@@ -5,7 +5,8 @@ ts_project(
     name = "backend",
     declaration = True,
     tsconfig = "//examples/isolated_declarations:tsconfig",
-    deps = ["//examples/isolated_declarations/core"],
+    # FIXME: should just be core:core
```

Comment:
@jbedard this is the next step in the PR, to make the SWC-transpiled declarations appear as the types on the terminal rule in the macro produced for `core:core`. I have to juggle the logic for when to produce a `js_library` target "tree".

Also before landing this I think the declaration producing rule should be in rules_swc rather than a bzl file here.